src/frontends/paddle/tests/paddle_pip_check.py

```diff
@@ -1,14 +1,56 @@
 import pkg_resources
 import re
 import sys
+import os
 
 req_file=sys.argv[1]
 
+constraints = {}
+constraints_path = None
+requirements = []
+
+# read requirements and find constraints file
+with open(req_file) as f:
+    raw_requirements = f.readlines()
+for line in raw_requirements:
+    if line.startswith("-c"):
+        constraints_path = os.path.join(os.path.dirname(req_file), line.split(' ')[1][:-1])
+
+# read constraints if they exist
+if constraints_path:
+    with open(constraints_path) as f:
+        raw_constraints = f.readlines()
+    for line in raw_constraints:
+        if line.startswith("#") or line=="\n":
+            continue
+        line = line.replace("\n", "")
+        package, delimiter, constraint = re.split("(~|=|<|>|;)", line, maxsplit=1)
+        if constraints.get(package) is None:
+            constraints[package] = [delimiter + constraint]
+        else:
+            constraints[package].extend([delimiter + constraint])
+
+    for line in raw_requirements:
+        if line.startswith(("#", "-c")):
+            continue
+        line = line.replace("\n", "")
+        if re.search("\W", line):
```

Comment:
I think searching for the delimiter like '~|=|<|>' here is better.

Reply:
For example, some python modules such as 'pip-versions' can also be recognized containing '\W'.

Reply:
@meiyang-intel is right, I think I need to update the parsers in `setup.py` files as well. Great catch :)

Reply:
Done